### PR TITLE
unzip the file from the binary URL before spawn

### PR DESCRIPTION
**Pull request by daichitominaga**
## Description

The URL for agent updates is the GitHub release URL. 
Updates are distributed there in ZIP format, requiring unzipping.

## Check
request is successful.
```
2024-02-01T20:54:35.374842260+09:00 [INFO] - actix_web::middleware::logger - - "POST /internal/version/update HTTP/1.1" 200 24 "-" "got (https://github.com/sindresorhus/got)" 0.956834 -
```

The process is spawned, but the original process remains. Therefore, it may be necessary to kill the process. (Since it is forced to work in a different environment, the updated process is crashing with an error.)
```
ps aux | grep nodex
163156  0.1  0.0 1863508 23724 pts/1   Sl+  20:47   0:00 target/debug/nodex-agent
181324  0.0  0.0      0     0 pts/1    Z+   20:54   0:00 [nodex-agent] <defunct>
181514  0.0  0.0  10080  2688 pts/0    S+   20:55   0:00 grep --color=auto nodex
```

### CODE REVIEW

File: tmp/nodex-agent-macos-latest.zip
Comment (rabe1028) [inline tmp/nodex-agent-macos-latest.zip:1]: @daichitominaga 
I thought you didn't need this file.

Comment (daichitominaga) [reply tmp/nodex-agent-macos-latest.zip:1]: sorry, I didn't notice the difference. I have deleted the relevant file.

File: src/lib.rs
Comment (rabe1028) [inline src/lib.rs:1]: @daichitominaga 
Q. What is this file for?

Comment (daichitominaga) [reply src/lib.rs:1]: @rabe1028 
sorry.removed.

**Review by rabe1028 — DISMISSED** [review]
LGTM

**Review by rabe1028 — APPROVED** [review]

**Pull request merged**